tests/e2e/101_rest_api_test.go

```diff
@@ -27,7 +27,7 @@ func TestInfoVersionHandler(t *testing.T) {
 	ctx := Setup(t)
 	defer ctx.Teardown()
 
-	version, err := ctx.agentClient.AgentVersion(context.Background())
+	version, err := ctx.Agent.client.AgentVersion(context.Background())
```

Comment:
The client should be accessed via an exported method to make the planned separation easier.